src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -279,35 +271,25 @@ export const handleEncryptedSubmission: RequestHandler = async (req, res) => {
   }
 
   // Save Responses to Database
-  // TODO(frankchn): Extract S3 upload functionality to a service
   const formData = req.body.encryptedContent
-  const attachmentData = req.body.attachments || {}
   const { verified } = res.locals
-  const attachmentMetadata = new Map()
-  const attachmentUploadPromises = []
-
-  // Object.keys(attachmentData[fieldId].encryptedFile) [ 'submissionPublicKey', 'nonce', 'binary' ]
-  for (const fieldId in attachmentData) {
-    const individualAttachment = JSON.stringify(attachmentData[fieldId])
-
-    const hashStr = crypto
-      .createHash('sha256')
-      .update(individualAttachment)
-      .digest('hex')
-
-    const uploadKey =
-      form._id + '/' + crypto.randomBytes(20).toString('hex') + '/' + hashStr
-
-    attachmentMetadata.set(fieldId, uploadKey)
-    attachmentUploadPromises.push(
-      AwsConfig.s3
-        .upload({
-          Bucket: AwsConfig.attachmentS3Bucket,
-          Key: uploadKey,
-          Body: Buffer.from(individualAttachment),
-        })
-        .promise(),
+  let attachmentMetadata = new Map()
+
+  if (req.body.attachments) {
+    const attachmentUploadResult = await uploadAttachments(
+      form._id,
+      req.body.attachments,
     )
+
+    if (attachmentUploadResult.isErr()) {
+      return res.status(StatusCodes.BAD_REQUEST).json({
+        message:
+          'Could not upload attachments for submission. For assistance, please contact the person who asked you to fill in this form.',
+        spcpSubmissionFailure: false,
+      })
```

Comment:
rather than hardcoding the message here, it might be better to add the new `AttachmentUploadError` to the `mapRouteError` function so we can view all the status codes and messages in one place